book/src/proposals/snapshot-verification.md

```diff
@@ -2,8 +2,41 @@
 
 ## Problem
 
-Snapshot verification of the account states is implemented, but the bank hash of the snapshot which is used to verify is falsifiable.
+When a validator boots up from a snapshot, it needs a way to verify the account set matches what the rest of the network sees quickly. A potential
+attacker could give the validator an incorrect state, and then try to convince it to accept a transaction that would otherwise be rejected.
 
 ## Solution
 
-While a validator is processing transactions to catch up to the cluster from the snapshot, use incoming vote transactions and the commitment calculator to confirm that the cluster is indeed building on the snapshotted bank hash. Once a threshold commitment level is reached, accept the snapshot as valid and start voting.
+Currently, the bank hash is derived from xor'ing an image derived from the hash of the
+account state and slot. Unfortunately this is insecure as an xor collision can be found
+with O(n) bits in the image and O(n^4) computation.
+
+We propose to use an RSA accumulator where we hash the account state to a large prime-ish number.
```

Comment:
Write it as if the proposal has been accepted.  We don’t need the old one referenced.